lua/telescope/builtin/__files.lua

```diff
@@ -162,10 +162,20 @@ files.live_grep = function(opts)
 end
 
 files.grep_string = function(opts)
-  -- TODO: This should probably check your visual selection as well, if you've got one
   opts.cwd = opts.cwd and vim.fn.expand(opts.cwd) or vim.loop.cwd()
   local vimgrep_arguments = vim.F.if_nil(opts.vimgrep_arguments, conf.vimgrep_arguments)
-  local word = vim.F.if_nil(opts.search, vim.fn.expand "<cword>")
+  local word
+  local visual = vim.fn.mode() == "v"
+
+  if visual == true then
+    local saved_reg = vim.fn.getreg("v")
+    vim.cmd([[sil norm "vy]])
```

Comment:
Please add `noautocmd` before `sil` to surpress `TextYankPost` event.

Reply:
Ok I did it.